openquakeplatform/openquakeplatform/static/js/irv_viewer.js

```diff
@@ -14,1043 +14,839 @@
+        '<option>IR</option>'
+    );
+
+    // Add SVI themes to selection menu
+    $('#thematic-map-selection').append('<optgroup label="SVI Indicators">');
+    for (var id = 0; id < allSVIThemes.length; id++) {
+        $('#thematic-map-selection').append('<option>'+allSVIThemes[id]+'</option>');
+    }
 
-            ////////////////////////////////////////
-            //// Apply the weights to the values ///
-            ////////////////////////////////////////
+    // Add IR themes to selection menu
+    $('#thematic-map-selection').append('<optgroup label="IR Indicators">');
+    for (var ie = 0; ie < allRiskIndicators.length; ie++) {
+        $('#thematic-map-selection').append('<option>'+allRiskIndicators[ie]+'</option>');
+    }
 
-            weightedAalIndicator = jQuery.extend(true, {}, aalIndicator);
+    // Add all primary indicators to selection menu
+    $('#thematic-map-selection').append('<optgroup label="Primary Indicators">');
+    for (var ic = 0; ic < allPrimaryIndicators.length; ic++) {
+        $('#thematic-map-selection').append('<option>'+allPrimaryIndicators[ic]+'</option>');
+    }
 
-            // Multiply the aal value by the weighted value
-            $.each(weightedAalIndicator, function(key, value) {
-                var aalValue = 0;
-                aalValue = (value * tempAalWeight);
-                weightedAalIndicator[key] = aalValue;
-            });
+    $('#thematic-map-selection').change(function() {
+        thematicMap(layerAttributes);
+    });
 
-            ///////////////
-            //// Scale ////
-            ///////////////
+    thematicMap(layerAttributes);
 
-            scaledAalIndicator = jQuery.extend(true, {}, aalIndicator);
+    IRI.plotElement = "iri"; // Lable within the object
+    RI.plotElement = "ri"; // Lable within the object
+    SVI.plotElement = "svi"; // Lable within the object
+    var iriPcpData = [];
+    iriPcpData.push(IRI);
+    iriPcpData.push(SVI);
+    iriPcpData.push(RI);
+    IRI_PCP_Chart(iriPcpData);
 
 
-            // Scale the pri values
-            var aalValueArray = [];
-            var scaleAALvalues = [];
+    ///////////////////////////////////
+    //// Compute the IRI Indicator ////
+    ///////////////////////////////////
 
-            for (var v in scaledAalIndicator) {
-                aalValueArray.push(scaledAalIndicator[v]);
-            }
+} // End processIndicators
 
-            var tempAALmin = Math.min.apply(null, aalValueArray),
-                tempAALmax = Math.max.apply(null, aalValueArray);
+function scale(IndicatorObj) {
+    var ValueArray = [];
+    var scaledValues = [];
+    for (var v in IndicatorObj) {
+        ValueArray.push(IndicatorObj[v]);
+    }
+    var tempMin = Math.min.apply(null, ValueArray),
+        tempMax = Math.max.apply(null, ValueArray);
+    for (var j = 0; j < ValueArray.length; j++) {
+        // make sure not to devide by zero
+        if (tempMax  == tempMin) {
+            ValueArray[j] = 1;
```

Comment:
Please add a comment to specify that `1` is an arbitrary choice to translate a flat array into an array where each element equals to 1 (we might ask Chris if he likes such decision or if he prefers 0, or 0.5, or else instead).